doc/api/n-api.md

```diff
@@ -0,0 +1,3150 @@
+                                    napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: The length in bytes of the array buffer to create
+- `[out] data`: Pointer to the underlying byte buffer of the ArrayBuffer
+- `[out] result`: A `napi_value` representing a JavaScript ArrayBuffer
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API returns an N-API value corresponding to a JavaScript ArrayBuffer.
+ArrayBuffers are used to represent fixed-length binary data buffers. They are 
+normally used as a backing-buffer for TypedArray objects.
+The ArrayBuffer allocated will have an underlying byte buffer whose size is 
+determined by the `length` parameter that's passed in. 
+The underlying buffer is optionally returned back to the caller in case the
+caller wants to directly manipulate the buffer. This buffer can only be 
+written to directly from native code. To write to this buffer from JavaScript,
+a typed array or DataView object would need to be created.
+
+JavaScript ArrayBuffer objects are described in 
+[Section 24.1](https://tc39.github.io/ecma262/#sec-arraybuffer-objects) 
+of the ECMAScript Language Specification.
+
+#### *napi_create_buffer*
+
+##### Signature
+```C
+napi_status napi_create_buffer(napi_env env,
+                               size_t size,
+                               void** data,
+                               napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  size`: Size in bytes of the underlying buffer
+- `[out] data`: Raw pointer to the underlying buffer
+- `[out] result`: A `napi_value` representing a `node::Buffer`
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API allocates a `node::Buffer` object. While this is still a 
+fully-supported data structure, in most cases using a TypedArray will suffice.
+
+#### *napi_create_buffer_copy*
+
+##### Signature
+```C
+napi_status napi_create_buffer_copy(napi_env env,
+                                    size_t length,
+                                    const void* data,
+                                    void** result_data,
+                                    napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  size`: Size in bytes of the input buffer (should be the same as the
+ size of the new buffer)
+- `[in]  data`: Raw pointer to the underlying buffer to copy from
+- `[out] result_data`: Pointer to the new Buffer's underlying data buffer
+- `[out] result`: A `napi_value` representing a `node::Buffer`
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API allocates a `node::Buffer` object and initializes it with data copied
+from the passed-in buffer. While this is still a fully-supported data 
+structure, in most cases using a TypedArray will suffice.
```

Comment:
It should probably be mentioned that in Node.js >= 4 (i.e. all currently supported versions) `Buffer`s *are* Uint8Arrays